packages/components/src/internal/components/auditlog/AuditQueriesListingPage.tsx

```diff
@@ -73,18 +76,19 @@ export class AuditQueriesListingPage extends PureComponent<Props, State> {
     setLastSelectedId = async (): Promise<void> => {
         if (!this.hasDetailView()) return;
 
-        const model = this.getQueryGridModel();
+        const model = this.getQueryModel();
+        if (!model) return;
 
         // if the model has already loaded selections, we can use that to reselect the last row
         // otherwise, query the server for the selection key for this model and use that response (issue 39374)
-        if (model.selectedLoaded) {
+        if (!model.isLoadingSelections) {
             this.updateSelectedRowId(this.getLastSelectedId());
-        } else {
+        } else if (!isLoading(model.queryInfoLoadingState)) {
             const response = await getSelected(
```

Comment:
This call to `getSelected` seems redundant given that you pass `loadSelections` as  true in the call to `addModel`. Is there a reason we need to load the selections a second time?

Reply:
Good call. I don't believe that call to `getSelected` is necessary either. I'll remove it.